core/trino-main/src/main/java/io/trino/sql/planner/plan/PlanNode.java

```diff
@@ -92,6 +94,11 @@ public PlanNodeId getId()
 
     public abstract List<Symbol> getOutputSymbols();
 
+    public Set<Symbol> getOutputSymbolsSet()
```

Comment:
Let's not add this kind of method. It pollutes the interface and adds more confusion

Reply:
The other solution would be to relax the return type of `getOutputSymbols` to `Collection<Symbol>` and make a copy to other type if needed. There will be no performance penalty since all implementations return list, and the old `ProjectNode` implementation did this copy anyway.
WDYT?

Reply:
That's the opposite direction of how we want to evolve PlanNode. Eventually, the "output" of a PlanNode should represent a `row(...)` value, so the order of the fields is significant. A Collection doesn't capture that.